api/modelmanager/modelmanager_test.go

```diff
@@ -403,6 +404,62 @@ func (s *modelmanagerSuite) TestModelStatusError(c *gc.C) {
 	c.Assert(out, gc.IsNil)
 }
 
+func (s *modelmanagerSuite) TestListModelsWithInfo(c *gc.C) {
+	userTag := names.NewUserTag("commander")
+	modelInfo := params.ModelInfo{
+		Name:               "name",
+		UUID:               "uuid",
+		ControllerUUID:     "controllerUUID",
+		ProviderType:       "aws",
+		DefaultSeries:      "xenial",
+		CloudTag:           "cloud-aws",
+		CloudRegion:        "us-east-1",
+		CloudCredentialTag: "credential-cred",
+		OwnerTag:           "user-admin",
+		Life:               params.Alive,
+		Status:             params.EntityStatus{Status: status.Status("active")},
+		Users:              []params.ModelUserInfo{},
+		Machines:           []params.ModelMachineInfo{},
+	}
+
+	apiCaller := basetesting.BestVersionCaller{
+		BestVersion: 4,
+		APICallerFunc: func(objType string, version int, id, request string, arg, result interface{}) error {
+			c.Check(objType, gc.Equals, "ModelManager")
+			c.Check(id, gc.Equals, "")
+			c.Check(request, gc.Equals, "ListModelsWithInfo")
+			c.Check(arg, gc.Equals, params.Entity{Tag: userTag.String()})
+			c.Check(result, gc.FitsTypeOf, &params.ModelInfoResults{})
+
+			out := result.(*params.ModelInfoResults)
+			out.Results = []params.ModelInfoResult{
+				params.ModelInfoResult{Result: &modelInfo},
+				params.ModelInfoResult{Error: common.ServerError(errors.New("model error"))},
+			}
+			return nil
+		},
+	}
+
+	client := modelmanager.NewClient(apiCaller)
+	results, err := client.ListModelsWithInfo(userTag)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(results, jc.DeepEquals, []params.ModelInfoResult{
+		{Result: &modelInfo},
+		{Error: &params.Error{Message: "model error"}},
+	})
+}
+
+func (s *modelmanagerSuite) TestListModelsWithInfoError(c *gc.C) {
+	apiCaller := basetesting.APICallerFunc(
+		func(objType string, version int, id, request string, args, result interface{}) error {
+			return errors.New("captain, error")
+		})
+	client := modelmanager.NewClient(apiCaller)
+	out, err := client.ListModelsWithInfo(names.NewUserTag("captain"))
+	c.Assert(err, gc.ErrorMatches, "captain, error")
+	c.Assert(out, gc.IsNil)
+}
+
```

Comment:
Should we have a test that we can fall back to the old versions?

Reply:
all other tests will check older versions. I think here, specifically, we just want to know that everything is ok with new version.. why would be want to drop to old versions here? isn't it a responsibility of the client?

Reply:
I'm not saying specifically here, more just a general "do we have a test around the fallbacks". If this is specifically the API layer and not CLI, then as long as we still have the tests that we can use the others its fine. I am curious if we'll want to emulate the old behavior inside api/ instead of putting all of that logic in cmd/*

Reply:
Let's merge this as-is and consider whether this is what we want to do later in the branch :)